src/pybind/mgr/dashboard/module.py

```diff
@@ -477,13 +477,23 @@ def _toplevel_data(self):
         class Root(EndPoint):
             @cherrypy.expose
             def filesystem(self, fs_id):
+                try:
+                    fs_id = int(fs_id)
+                except ValueError:
+                    raise cherrypy.HTTPError(400,
+                        "Invalid filesystem id {0}".format(fs_id))
+
                 template = env.get_template("filesystem.html")
 
                 toplevel_data = self._toplevel_data()
 
-                content_data = {
-                    "fs_status": global_instance().fs_status(int(fs_id))
-                }
+                try:
+                    content_data = {
+                        "fs_status": global_instance().fs_status(fs_id)
+                    }
+                except TypeError:
```

Comment:
Catching TypeError is kind of weird here, better to change fs_status to raise a sane exception: (done in https://github.com/ceph/ceph/pull/18039/files)